src/easyscience/Objects/variable/descriptor_array.py

```diff
@@ -0,0 +1,760 @@
+
+            # Update variances as the square of the errors
+            self._array.variances = error**2
+        else:
+            self._array.variances = None
+
+    def convert_unit(self, unit_str: str) -> None:
+        """
+        Convert the value from one unit system to another.
+
+        :param unit_str: New unit in string form
+        """
+        if not isinstance(unit_str, str):
+            raise TypeError(f'{unit_str=} must be a string representing a valid scipp unit')
+        try:
+            new_unit = sc.Unit(unit_str)
+        except UnitError as message:
+            raise UnitError(message) from None
+
+        # Save the current state for undo/redo
+        old_array = self._array
+
+        # Perform the unit conversion
+        try:
+            new_array = self._array.to(unit=new_unit)
+        except Exception as e:
+            raise UnitError(f"Failed to convert unit: {e}") from e
+
+        # Define the setter function for the undo stack
+        def set_array(obj, scalar):
+            obj._array = scalar
+
+        # Push to undo stack
+        self._global_object.stack.push(
+            PropertyStack(self, set_array, old_array, new_array, text=f"Convert unit to {unit_str}")
+        )
+
+        # Update the array
+        self._array = new_array
+
+    def __copy__(self) -> DescriptorArray:
+        """
+        Return a copy of the current DescriptorArray.
+        """
+        return super().__copy__()
+
+    def __repr__(self) -> str:
+        """
+        Return a string representation of the DescriptorArray, showing its name, value, variance, and unit.
+        Large arrays are summarized for brevity.
+        """
+        # Base string with name
+        string = f"<{self.__class__.__name__} '{self._name}': "
+
+        # Summarize array values
+        values_summary = np.array2string(
+            self._array.values, 
+            precision=4, 
+            threshold=10,  # Show full array if <=10 elements, else summarize
+            edgeitems=3,   # Show first and last 3 elements for large arrays
+        )
+        string += f"values={values_summary}"
+
+        # Add errors if they exists
+        if self._array.variances is not None:
+            errors_summary = np.array2string(
+                self.error, 
+                precision=4, 
+                threshold=10, 
+                edgeitems=3,
+            )
+            string += f", errors={errors_summary}"
+
+        # Add unit
+        obj_unit = str(self._array.unit)
+        if obj_unit and obj_unit != "dimensionless":
+            string += f", unit={obj_unit}"
+
+        string += ">"
+        string=string.replace('\n', ',')
+        return string    
+
+
+    def as_dict(self, skip: Optional[List[str]] = None) -> Dict[str, Any]:
+        """
+        Dict representation of the current DescriptorArray.
+        The dict contains the value, unit and variances, in addition
+        to the properties of DescriptorBase.
```

Comment:
our `ruff` specs allow lines up to 127 characters. No need to split unnecessarily.